### ISSUE TITLE
setup-pulumi action uses deprecated version of node

### ISSUE THREAD

**Issue by ianschmitz** [pulumi/setup-pulumi#36]
### What happened?

Running a job that uses `pulumi/setup-pulumi@v2` results in the following warning from GitHub Actions:

> Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: pulumi/setup-pulumi

### Steps to reproduce

Run a job using the action

### Expected Behavior

The action uses a supported version of Node

### Actual Behavior

Node 12 actions are deprecated

### Output of `pulumi about`

N/A

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by Zaid-Ajaj** [comment]
Hi @ianschmitz this issue will be resolved when https://github.com/pulumi/setup-pulumi/pull/35 is merged which is currently blocked by https://github.com/pulumi/actions/pull/753

**Comment by RobbieMcKinstry** [comment]
GH has delayed the deprecation of GHE v3.3 until 18 Jan 2023. As a result, that's the new "do not merge until" date for this PR. :(

**Comment by aniravi24** [comment]
This should be unblocked now, Pulumi has merged this in!

**Comment by RobbieMcKinstry** [comment]
Indeed, it's unblocked! 🎉

**Comment by alexandre-girard-maif** [comment]
any news about this issue ?

**Comment by cobraz** [comment]
AFAIK, the error should disappear when updating to v4. Let me know if there's still issues :)

**Comment by aniravi24** [comment]
Sorry @cobraz , v4 of what? This GitHub action is on v2. I thought the resolution for this issue is merging [this](https://github.com/pulumi/setup-pulumi/pull/35/files) PR and making sure this action is updated on marketplace.

I might be missing something, but I don't think this issue is resolved yet.

**Comment by cobraz** [comment]
You are right! I got mixed up with the pulumi/actions.

**Comment by andriiholubenko** [comment]
Hi, do you have any updates?

**Comment by RobbieMcKinstry** [comment]
Hi @andriiholubenko , not at this time, no. We're currently pulled in other directions. I expect I can find time this quarter, but I can't give any better estimates than that currently. I'm sorry, I wish I had a better answer for you.

**Comment by joeduffy** [comment]
@abhinav, given https://github.com/pulumi/setup-pulumi/pull/35#issuecomment-1462383317, is this issue still relevant? It says that `setup-pulumi` is deprecated?

**Comment by abhinav** [comment]
@joeduffy Yeah, good call. This issue is no longer relevant since https://github.com/pulumi/setup-pulumi/pull/35#issuecomment-1462383317.

---

Folks, as mentioned in https://github.com/pulumi/setup-pulumi/pull/35#issuecomment-1462383317, please switch to pulumi/actions with the following single line change:


<table>
<thead><tr><td>Before</td><td>After</td></tr></thead>
<tbody>
<tr><td>

```yaml
- name: Install pulumi
  uses: pulumi/setup-pulumi@v2
```

---

```yaml
- name: Install pulumi
  uses: pulumi/setup-pulumi@v2
  with:
    pulumi-version: 3.3.0
```

</td><td>

```yaml
- name: Install pulumi
  uses: pulumi/actions@v4
```

---

```yaml
- name: Install pulumi
  uses: pulumi/actions@v4
  with:
    pulumi-version: 3.3.0
```

</td></tr>

</tbody>
</table>


Thanks!

Closing this issue as not planned.
